debuggers/gdb/gdb_mi_driver.py

```diff
@@ -29,57 +29,100 @@ def __init__(self, base_args, base_script_file_path, regression_args, regression
+        raw_result += self.regressed_gdb_instance.recv()
+        
+        # parse output regression
+        regressed_response = self.parse_command_output(raw_result)
 
         return { "base": base_response, "regressed": regressed_response }
 
+    def parse_command_output(self, raw_result):
+        response = []
+        for item in raw_result:
+            if item['type'] == 'console':
+                input_string = str(item['payload'])
+                processed_output = parse_gdb_line(input_string)
+                response.append(processed_output)
+        return response
+    
     def run_single_command(self, command, version):
         global base_response
         global regressed_response
         
-        result = []
-        raw_result = self.gdb_instances[version].write(" {command}\n".format(command = command), 2)
+        self.gdb_instances[version].send(((" {command}\n".format(command = command),), {"timeout_sec": 60}))
+        raw_result = self.gdb_instances[version].recv()
         
         # make sure all output is flushed
         # time.sleep(.005)
-        flushed_results = self.gdb_instances[version].write("".format(command = command))
-        raw_result = raw_result + flushed_results
-        
-        for item in raw_result:
-            if item['type'] == 'console':
-                input_string = str(item['payload'])
-                processed_output = parse_gdb_line(input_string)
-
-                result.append(processed_output)
+        self.gdb_instances[version].send((("",), {"timeout_sec": 60}))
+        raw_result += self.gdb_instances[version].recv()
         
-        return result
+        return self.parse_command_output(raw_result)
 
     def run_single_special_command(self, command, version):
         global base_response
         global regressed_response
 
-        raw_result = self.gdb_instances[version].write(" {command}\n".format(command = command), 2)
-        flushed_results = self.gdb_instances[version].write("".format(command = command))
-        raw_result = raw_result + flushed_results
+        self.gdb_instances[version].send(((" {command}\n".format(command = command),), {"timeout_sec": 60}))
+        raw_result = self.gdb_instances[version].recv()
+
+        # flush output
+        self.gdb_instances[version].send((("",), {"timeout_sec": 60}))
+        raw_result += self.gdb_instances[version].recv()
 
+        return self.parse_special_command_output(raw_result)
+
+    def parse_special_command_output(self, raw_result):
         for item in raw_result:
             if item['type'] == 'console':
                 input_string = str(item['payload'])
                 processed_output = parse_gdb_line(input_string)
-                parsed_dict = json.loads(processed_output)
+                try:
+                    parsed_dict = json.loads(processed_output)
+                except json.JSONDecodeError:
+                    parsed_dict = processed_output
```

Comment:
This change is related to the `JSONDecodeError` we were talking about.
Without this change, I am getting the JSON error.

Reply:
Maybe there are certain cases where the processed_output is not in the correct format and json.loads fails to parse it.